src/server/server_family.cc

```diff
@@ -888,6 +888,10 @@ GenericError RewriteConfigFile() {
   return {};
 }
 
+bool IsMaster() {
+  return ServerState::tlocal() && ServerState::tlocal()->is_master;
```

Comment:
I assume server state should never be null under normal conditions?

Reply:
And that's why the tests failed. When we call `LoadFromSnapshot()` when df starts, `ServerState::tlocal()` is `0x00`. I patched `IsMaster()` to reflect this. 

So yes, we need `ServerState::tlocal()` or otherwise we might access a nullptr.

Imo, if this is an initialization issue I will follow up in a separate PR and clean that up as well.